### DISCUSSION TITLE
Which map do you like best?

### DISCUSSION THREAD

**Discussion by TheMowgliMan** [TheMowgliMan/Country-Clash#5] [Polls]
I'm making a few different maps. Pick the most realistic one please.

Map 1:
![Country Clash- The Game (DEBUG) 5_22_2022 1_01_23 PM](https://user-images.githubusercontent.com/96443213/169706905-088d7574-76b7-4c09-83ca-c576d1cdfc77.png)
        terrain_noise.octaves = 5
	terrain_noise.period = 128
	terrain_noise.persistence = 0.8
	terrain_noise.lacunarity = 3

Map 2:
![Country Clash- The Game (DEBUG) 5_22_2022 1_03_45 PM](https://user-images.githubusercontent.com/96443213/169706959-40b4a4f1-c03b-41e0-a0fd-5ed5765ce295.png)
	terrain_noise.octaves = 5
	terrain_noise.period = 110
	terrain_noise.persistence = 0.7
	terrain_noise.lacunarity = 2

Map 3:
![Country Clash- The Game (DEBUG) 5_22_2022 1_05_14 PM](https://user-images.githubusercontent.com/96443213/169707009-60513b53-129f-4f97-bc4b-a136b1dfadc3.png)
	terrain_noise.octaves = 5
	terrain_noise.period = 128
	terrain_noise.persistence = 0.6
	terrain_noise.lacunarity = 2.5

**Comment by TheMowgliMan** [comment]
Note: I can also use Perlin noise to change these values back and forth to increase realism.

**Comment by 3quarterscolon** [comment]
Nice. It should be pretty easy to change the colors more based on height. It’s still not super realistic, though. It might be a fun side-thing to make more realistic terrain generation. One question, though: are we procedurally generating maps? Why not use the original maps?

**Reply by TheMowgliMan** [reply]
Having both pre-built and procedural maps would be good ideas. However, as we don't currently have a map editor, procedural is the only one for testing purposes.

**Reply by 3quarterscolon** [reply]
Okay, thanks.

**Comment by 3quarterscolon** [comment]
Suggestion: Make the water level higher. In real life, most of the Earth is covered in water.

**Reply by 3quarterscolon** [reply]
I can probably actually do this on my iPad, but I wouldn’t be able to see what it looks like so I wouldn’t be able to fine-tune it properly.

**Reply by TheMowgliMan** [reply]
Alright.

**Reply by TheMowgliMan** [reply]
I've adjusted it slightly:
![Country Clash- The Game (DEBUG) 5_22_2022 2_26_56 PM](https://user-images.githubusercontent.com/96443213/169710146-1ac58b16-ff0e-4724-8916-f98accb9eee9.png)

**Reply by 3quarterscolon** [reply]
That looks better. Maybe a little farther, but I think it’s up to personal opinion more than anything. I think the biggest problem now is that the map isn’t seamless.

**Reply by TheMowgliMan** [reply]
The map has a set size. For quick debugging purposes, it's at 720*720, which is smaller that the window size, so you see some background. For actual gameplay, it'll be 4096 * 4096.

**Reply by TheMowgliMan** [reply]
I wish it could be bigger, but windows kills it if it's bigger.

**Reply by 3quarterscolon** [reply]
> The map has a set size. For quick debugging purposes, it's at 720*720, which is smaller that the window size, so you see some background. For actual gameplay, it'll be 4096 * 4096.

No, I mean if you put the map next to itself, there’s an obvious seam. This means that if a unit travels from one side of the map to the other, the terrain will sharply jump from one type to another.

**Reply by TheMowgliMan** [reply]
Ahhhh.... I see. We can add 4 more sprites (maps) each holding the same texture, so when someone get's to the edge of the map it jumps over back to the middle. It won't be visible and hardly tax more resources.

**Reply by 3quarterscolon** [reply]
> We can add 4 more sprites (maps) each holding the same texture, so when someone get's to the edge of the map it jumps over back to the middle.

I’m sorry, I don’t quite understand what you mean by this. Can you clarify?

**Reply by TheMowgliMan** [reply]
Duplicate the map 4 or 8 times (depends) and then put a duplicate to the edge of the primary map, and jump to the other side of the map when the player camera reaches there. We could also just have one map, and add some portal lookin thingies and that way avoid some difficult code. Or we can add *mysterious* barriers and simply prevent jumping over to the other side of the map.

Better looking map, by the way:
![Country Clash- The Game (DEBUG) 5_22_2022 2_59_07 PM](https://user-images.githubusercontent.com/96443213/169711547-6daf0585-ea80-4e7d-bf67-b84615254b67.png)

**Reply by 3quarterscolon** [reply]
> Duplicate the map 4 or 8 times (depends) and then put a duplicate to the edge of the primary map, and jump to the other side of the map when the player camera reaches there. We could also just have one map, and add some portal lookin thingies and that way avoid some difficult code. Or we can add _mysterious_ barriers and simply prevent jumping over to the other side of the map.

Okay. What I meant was that the map doesn’t loop properly. If you use an image editor and put one instance of the map next to another, you can see a sharp shoreline. The reason for this is that the Perlin noise is not tileable. You can create tileable Perlin noise, but right now it isn’t. I can illustrate this in an image if you want.

> Better looking map, by the way: ![Country Clash- The Game (DEBUG) 5_22_2022 2_59_07 PM](https://user-images.githubusercontent.com/96443213/169711547-6daf0585-ea80-4e7d-bf67-b84615254b67.png)

Nice. All that’s left now is mountains and beaches and stuff, which you could probably just make by adding new tiles types.

**Reply by TheMowgliMan** [reply]
Added Beaches (Will look better when I add zooming in):
![Country Clash- The Game (DEBUG) 5_23_2022 5_48_36 AM](https://user-images.githubusercontent.com/96443213/169822949-282be17d-5b33-4ef0-b41f-784e7475b8a3.png)

**Reply by 3quarterscolon** [reply]
I'm getting set up to start working now!

**Reply by 3quarterscolon** [reply]
How do you use github with this?

**Reply by 3quarterscolon** [reply]
I’ve started work on the main menu. You can see my progress on my fork. It’s not much, but I’m also rather new to this.

**Reply by TheMowgliMan** [reply]
Alrighty- I'll add source code install instructions for installing the game in the README.

**Reply by TheMowgliMan** [reply]
Just letting you know, you can work on the main branch directly without PRs or forks. That's why I added you as a collaborator.

**Reply by 3quarterscolon** [reply]
> Just letting you know, you can work on the main branch directly without PRs or forks. That's why I added you as a collaborator.

I know, but I don’t want to start adding stuff and then have to leave those changes unfinished until late the next day or whatever. If it’s something small, I’ll edit the main repo instead.

**Comment by TheMowgliMan** [comment]
Yeah. I'll add a little better map storage design later.

**Reply by TheMowgliMan** [reply]
Right now all it stores are textures, I'll make it store tiles and then convert it to textures. It _should_ look the same.

**Comment by TheMowgliMan** [comment]
Right 👍
